wpilibj/src/main/java/edu/wpi/first/wpilibj/sysid/SysIdRoutineLog.java

```diff
@@ -221,6 +219,9 @@ public MotorLog motor(String motorName) {
    * @param state The current state of the SysId test routine.
    */
   public void recordState(State state) {
+    if (m_state == null) {
+      m_state = new StringLogEntry(DataLogManager.getLog(), "sysid-test-state-" + m_logName);
```

Comment:
Why does C++ append kNone and Java not?

Reply:
Fixed, kNone isn't supposed to be logged in either language